robot_controllers/src/diff_drive_base.cpp

```diff
@@ -293,7 +293,17 @@ void DiffDriveBaseController::update(const rclcpp::Time& now, const rclcpp::Dura
     r = std::max(-max_velocity_r_, std::min(actual_scaling * desired_r_, max_velocity_r_));
   }
 
-  double elapsed = dt.nanoseconds() / 1e9;
+  // Make sure "dt" is never negative, and warn if it is 0.0 (since it really shouldn't be)
+  double elapsed = dt.nanoseconds() / 1e9;  // seconds
+  if (elapsed <= 0.0)
+  {
+    RCLCPP_WARN(rclcpp::get_logger(getName()),
+                "BaseController - bad dt = %f", dt);
```

Comment:
Not sure if this will print the log source (i.e. base controller) twice. The ROS 1 version does not, which is why I explicitly had it in the message.

Reply:
Oh, noted. I had assumed it was printing twice in ROS1, too .